aptos-move/move-examples/mint_nft/step1/sources/nft_step1.move

```diff
@@ -0,0 +1,144 @@
+    BUILDING Examples
+    package size 2770 bytes
+    Do you want to submit a transaction for a range of [1164400 - 1746600] Octas at a gas unit price of 100 Octas? [yes/no] >
+    yes
+    {
+      "Result": {
+        "transaction_hash": "0x576a2e9481e71b629335b98ea75c87d124e1b435e843e7a2ef8938ae21bebfa3",
+        "gas_used": 11679,
+        "gas_unit_price": 100,
+        "sender": "a911e7374107ad434bbc5369289cf5855c3b1a2938a6bfce0776c1d296271cde",
+        "sequence_number": 0,
+        "success": true,
+        "timestamp_us": 1669659103283876,
+        "version": 12735152,
+        "vm_status": "Executed successfully"
+      }
+    }
+    */
+/// - 2.c Check the module we just published on the Aptos Explorer.
+/// Go to https://explorer.aptoslabs.com/. For the example above, we used devnet, so we will choose `devnet` at the
+/// top right of the screen and paste the transaction_hash `0x576a2e9481e71b629335b98ea75c87d124e1b435e843e7a2ef8938ae21bebfa3`
+/// in the `Search Explorer` box. We can see the changes we made by publishing this module under the `Changes` tab.
+///
+/// 3. Check out the `mint_nft()` function below - we are not going to run a command to mint the NFT in this step, because
+/// this function right now asks for two signers and that's inpractical to do using CLI commands.
+/// In the next step of this tutorial, we will introduce a way to programmatically sign for transactions, so the module publisher
+/// doesn't need to manually sign transactions, and only need one signer (the nft receiver's signer) for the `mint_nft()` function.
+module mint_nft::nft_step1 {
+    use std::error;
+    use std::string;
+    use std::vector;
+
+    use aptos_token::token;
+    use std::signer;
+    use std::string::String;
+    use aptos_token::token::TokenDataId;
+
+    // This struct stores an NFT collection's relevant information
+    struct ModuleData has key {
+        token_data_id: TokenDataId,
+    }
+
+    /// Action not authorized because the signer is not the admin of this module
+    const ENOT_AUTHORIZED: u64 = 1;
+
+    /// `init_module` is automatically called when publishing the module.
+    /// In this function, we create an example NFT collection and an example token.
+    fun init_module(source_account: &signer) {
+        let collection_name = string::utf8(b"Collection name");
+        let description = string::utf8(b"Description");
+        let collection_uri = string::utf8(b"Collection uri");
+        let token_name = string::utf8(b"Token name");
+        let token_uri = string::utf8(b"Token uri");
+        let maximum_supply = 0;
+        let mutate_setting = vector<bool>[ false, false, false ];
+
+        // Create the nft collection.
+        token::create_collection(source_account, collection_name, description, collection_uri, maximum_supply, mutate_setting);
+
+        // Create a token data id to specify the token to be minted.
+        let token_data_id = token::create_tokendata(
```

Comment:
So this collection only has a single token type? I wonder if we can add a step 2 first (and make current step 2 step 3) to introduce a new type of token and mint it as well. This would however require changing mint_nft function to allow specifying which token to mint.

Reply:
ahh yeah, so we will have that soon, probably in another tutorial. this is just an example of how to create something like our Aptos testnet NFT / an event pass NFT, where we would have the same token data id and picture for every token minted